Data/Root/RootDataIO.hpp

```diff
@@ -29,8 +29,9 @@ class RootDataIO {
   /// \param TreeName_	Name of tree in input or output file
   /// \param NumberEventsToProcess_	-1 processes all events
   RootDataIO(const std::string TreeName_ = "data",
-             int NumberEventsToProcess_ = -1);
+             std::size_t NumberEventsToProcess_ = -1);
```

Comment:
I just noticed this. It will work because it will set it to the max unsigned number. If we would use the max value for `size_t`, it would be a bit more transparent. `std::numeric_limits<size_t>::max()` should do it https://en.cppreference.com/w/cpp/types/numeric_limits. Then you could also remove the `NumberEventsToProcess <= 0` in the above if statement